src/rust/rust-proto/src/graplinc/grapl/common/v1beta1/types.rs

```diff
@@ -0,0 +1,245 @@
+use crate::{
+    protobufs::graplinc::grapl::common::v1beta1::{
+        EdgeName as EdgeNameProto,
+        NodeType as NodeTypeProto,
+        PropertyName as PropertyNameProto,
+        Uid as UidProto,
+    },
+    serde_impl,
+    type_url,
+    SerDeError,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct PropertyName {
+    pub value: String,
+}
+
+impl std::fmt::Display for PropertyName {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.value)
+    }
+}
+
+impl PropertyName {
+    pub fn new_unchecked(value: String) -> Self {
+        // todo: debug assertion
+        Self { value }
+    }
+}
+
+impl TryFrom<&'static str> for PropertyName {
+    type Error = SerDeError;
+    fn try_from(raw: &'static str) -> Result<Self, Self::Error> {
+        if raw.is_empty() {
+            return Err(SerDeError::InvalidField {
+                field_name: "PropertyName",
+                assertion: "can not be empty".to_owned(),
+            });
+        }
+        if raw.len() > 32 {
+            return Err(SerDeError::InvalidField {
+                field_name: "PropertyName",
+                assertion: "can not be more than 32 characters".to_owned(),
+            });
+        }
+
+        Ok(Self {
+            value: raw.to_owned(),
+        })
+    }
+}
+
+impl TryFrom<String> for PropertyName {
+    type Error = &'static str;
```

Comment:
Shouldn't this also be a `SerDeError`, in keeping with the above `TryFrom` implementation?

This actually goes for all such conversion traits in this file... what determines that one has an Error type of `SerDeError`, while another uses `&'static str`?

Reply:
The reason it's not a SerDeError is because it's not deserializing anything. The SerDeError that would be appropriate is the SerDeError::InvalidField, which requires the field name, which is not provided.

Reply:
this should def be serde error

Reply:
@colin-grapl By that logic, `impl TryFrom<&'static str> for PropertyName` shouldn't have `SerDeError` as its error type either.

My broader point is that there's really nothing materially different, on a business logic level, between converting from a `&str` and from a `String`, so I'd expect these implementations to look effectively identical. Having small differences like this just make things more confusing.

Reply:
Yes, I think that they both should not return SerDeError. You can see in the case where it does that the field_name is just "PropertyName", which is not correct.

Reply:
Colin, unfortunately you're shoving too much validation logic into something that should just be a ser/de layer